zenodo/modules/records/serializers/json.py

```diff
@@ -109,3 +110,24 @@ def serialize_exporter(self, pid, record):
         return json.dumps(
             self.transform_search_hit(pid, record)
         ).encode('utf8')  + b'\n'
+
+    def serialize_search(self, pid_fetcher, search_result, links=None,
+                         item_links_factory=None, **kwargs):
+        """Serialize Zenodo search results and aggregations."""
+        aggregations = search_result.get('aggregations', dict())
+        if 'communities' in aggregations:
+            comm_aggs = aggregations['communities']['buckets']
+            keys = []
+            for bucket in comm_aggs:
+                keys.append(bucket['key'])
+            comms = Community.query.filter(Community.id.in_(keys)).all()
+            for comm in comms:
+                for bucket in comm_aggs:
+                    if bucket['key'] == comm.id:
+                        bucket['title'] = comm.title
```

Comment:
This aggregation "enhancing" logic specific to communities could be separated into another function, so that we can easily add similar enhancements in the future. Furthermore, it might be interesting to use something like `functools.lru_cache` to cache some of the results and minimize to some extent DB calls (imagine someone paginating through multiple pages of results)